### ISSUE TITLE
Warning: "CannotInitializeManagedResource" after creating RDSInstance

### ISSUE THREAD

**Issue by benjamin-hg** [crossplane/provider-aws#758]
### What happened?

I'm doing a PoC with the following RDS instance definition:

```yaml
apiVersion: database.aws.crossplane.io/v1beta1
kind: RDSInstance
metadata:
  name: poc-db-3
spec:
  forProvider: 
    dbInstanceClass: 'db.m5.large' 
    masterUsername: root
    allocatedStorage: 20 
    engine: postgres
    engineVersion: "13.2"
    storageEncrypted: true
    skipFinalSnapshotBeforeDeletion: true 
  writeConnectionSecretToRef:
      name: poc-postgres-secret-3
      namespace: crossplane-system
  providerConfigRef:
    name: default
  deletionPolicy: Delete 
```

**when doing**
`kubectl apply -f ...`

and waiting a few seconds, I see the following events for this resource:

```
Events:
  Type     Reason                           Age                 From                                            Message
  ----     ------                           ----                ----                                            -------
  Normal   CreatedExternalResource          5m4s                managed/rdsinstance.database.aws.crossplane.io  Successfully requested creation of external resource
  Warning  CannotInitializeManagedResource  62s (x2 over 5m4s)  managed/rdsinstance.database.aws.crossplane.io  cannot update RDS instance custom resource: Operation cannot be fulfilled on rdsinstances.database.aws.crossplane.io "poc-db-3": the object has been modified; please apply your changes to the latest version and try again
```

No further events happen and in AWS RDS console I don't see any instance.

**Expected:**
No "CannotInitializeManagedResource" warning and an actual RDS instance spinning up in AWS console.

### How can we reproduce it?

See steps above.

### What environment did it happen in?
Crossplane version: 

Chart Version: 1.3.0
Chart Application Version: 1.3.0

```
kubectl version 
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.7", GitCommit:"1dd5338295409edcfff11505e7bb246f0d325d15", GitTreeState:"clean", BuildDate:"2021-01-13T13:23:52Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.8-eks-96780e", GitCommit:"96780e1b30acbf0a52c38b6030d7853e575bcdf3", GitTreeState:"clean", BuildDate:"2021-03-10T21:32:29Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by hasheddan** [comment]
@benjamin-hg this error shouldn't cause concern as it is automatically retried and is just due to object version mismatch with the API server. I am surprised to hear that an RDS instance did not appear in the console. How long did you wait after seeing this error before deleting the object (if you have at all yet)?

**Comment by benjamin-hg** [comment]
@hasheddan thanks for the quick reply!
Good to know that this message doesn't indicate an error.
I'm working with two different AWS accounts (in different regions) and I possibly did a mistake configuring the provider credentials (messing with `AWS_PROFILE` env var). After ensuring the credentials of the desired account have been configured, the RDS instance appeared in the console soon after. 
Interestingly, I had the suspicion before and did check the other account as well -- but did not see any RDS instance there either -- but my resource got updated with a database host name of the region of my cluster. So it looked like I've spawned an RDS instance flying under the radar. But I'm not digging into it now -- happy that it works 🎉
